lib/Doctrine/DBAL/Connection.php

```diff
@@ -193,8 +195,20 @@ public function __construct(
         $this->_driver = $driver;
         $this->params  = $params;
 
+        // check for existing pdo object
+        if (isset($params['pdo']) && ! $params['pdo'] instanceof PDO) {
```

Comment:
This condition partially duplicates the following one. Would it make sense to deduplicate them?
```php
if (isset(...)) {
    Deprecation::trigger()

    if (! instanceof PDO) {
        throw
    }
}
```

Reply:
I'm all for changing it however it is wanted. The above code was a literal copy/paste from DriverManager. I wasn't sure if it was written like that to avoid nesting an `if` inside another `if` or what, so I kept it the same.

I swapped it to be like the above example, though.